addon/model-data.js

```diff
@@ -696,9 +727,10 @@ export default class M3ModelData {
   }
 
   _notifyRecordProperties(changedKeys) {
-    if (this._embeddedInternalModel) {
+    if (this._embeddedInternalModel && this._embeddedInternalModel.record) {
       this._embeddedInternalModel.record._notifyProperties(changedKeys);
-    } else {
+    } else if (!this._parentModelData) {
+      // only notify through the store if it is not a child model data
```

Comment:
if there is no embedded internal model this cannot be an embedded (child) model data.

let's keep these conditions simpler and instead add asserts

Reply:
The flow is the following:
- a `Company` projection named `MiniCompany`  with `address` field projected as `CountryOnlyAddress`.
- when we read `address` from the projection, it will create two model datas:
  - one in the context of `MiniCompany` model data, with a type `CountryOnlyAddress`
  - one in the context of `Company` model data, with type `Address`.
 However, we only receive an embedded internal model for the `CountryOnlyAddress` model data, because the `get()` is done for the projection only. We will populate the `embeddedInternalModel` of the base `Address` model data, when the code calls `get('address')` on the base `Company` model.